src/panels/logbook/ha-logbook-data.js

```diff
@@ -33,41 +45,71 @@ class HaLogbookData extends PolymerElement {
 
   hassChanged(newHass, oldHass) {
     if (!oldHass && this.filterDate) {
-      this.filterDateChanged(this.filterDate);
+      this.updateData();
     }
   }
 
-  filterDateChanged(filterDate) {
+  filterDataChanged(newValue, oldValue) {
+    if (oldValue !== undefined) {
+      this.updateData();
+    }
+  }
+
+  updateData() {
     if (!this.hass) return;
 
     this._setIsLoading(true);
 
-    this.getDate(filterDate).then(function (logbookEntries) {
-      this._setEntries(logbookEntries);
-      this._setIsLoading(false);
-    }.bind(this));
+    this.getDate(this.filterDate, this.filterPeriod, this.filterEntity)
+      .then(function (logbookEntries) {
+        this._setEntries(logbookEntries);
+        this._setIsLoading(false);
+      }.bind(this));
   }
 
-  getDate(date) {
-    if (!DATE_CACHE[date]) {
-      DATE_CACHE[date] = this.hass.callApi('GET', 'logbook/' + date).then(
-        function (logbookEntries) {
-          logbookEntries.reverse();
-          return logbookEntries;
-        },
-        function () {
-          DATE_CACHE[date] = false;
-          return null;
-        }
-      );
+  getDate(date, period, entityId) {
+    if (!period) period = DEFAULT_PERIOD;
```

Comment:
The data element should not handle defaults, defaults should be handled by the business logic that controls this element.

Reply:
On a side note, we should just kill this element and make it a JavaScript function.